src/main/java/org/isf/exatype/rest/ExamTypeController.java

```diff
@@ -71,8 +71,8 @@ public ResponseEntity<ExamTypeDTO> newExamType(@RequestBody ExamTypeDTO newExamT
 
 		ExamType examType = examTypeMapper.map2Model(newExamType);
 		ExamType createdExamType = examTypeBrowserManager.newExamType(examType);
-		if (createdExamType == null) {
-			throw new OHAPIException(new OHExceptionMessage("Exam Type type not created."));
+		if (!examTypeBrowserManager.isCodePresent(examType.getCode())) {
+			throw new OHAPIException(new OHExceptionMessage("Exam Type not created."));
```

Comment:
Why we check the existance of `createdExamType` with a new query on the DB rather than just check if the returning object is `null`?

Reply:
Because the contract for the underlying `save` method in the CRUD repository says:
<pre>
 @return the saved entity; will never be {@literal null}
</pre>

Reply:
Ok, right. But so maybe we don't need this check anymore? The entity is either saved (and returned) or an exception is thrown (and catched)?

Reply:
Ok, I will test the catch and see what to do in another issue probably. Thanks!